### ISSUE TITLE
allow specifying application to generate artifacts for

### ISSUE THREAD

**Issue by containscafeine** [surajssd/kapp#87]
Resulting from https://github.com/surajssd/kapp/issues/17#issuecomment-310578854

Now we support defining multiple applications in a single file (#86), and right now Kubernetes artifacts get generated for every application defined in that one file.

However, we should add support for choosing the application to generate artifacts for using the command line.

**Comment by containscafeine** [comment]
How do we want to proceed with this?

One way is to do something like -
- `kedge -f wordpress.yml generate`
- `kedge -f wordpress.yml generate web`
- `kedge -f wordpress.yml generate web db`

We should also support some way of excluding an app and even wildcards, so
- `kedge -f wordpress.yml generate we*`
- `kedge -f wordpress.yml generate !web`

**Comment by pradeepto** [comment]
@containscafeine I made some edits to your comment to make it a bit more explicit of what we are trying to do. Please review it once.

LGTM from my side. My suggestion is, let's always make sure that the UX is "simple". I like the wildcards idea but I am not sure how it would affect the UX and the implementation. How about we do the first part cleanly and get feedback first? WDYT?

Also, a major requirement in terms of UX: Error out when the user provides an incorrect service name is provided by the user. For example,  foobar service is not defined in wordpress.yml

`kedge -f wordpress.yml generate foobar` 
`FATAL: foobar - Incorrect or missing service name in wordpress.yml`

**Comment by containscafeine** [comment]
Okay, let's do this in 2 parts!

**Comment by surajssd** [comment]
`kedge -f wordpress.yml generate web`

This means that `web` is sub-command to generate, not sure if that is even possible to do with `cobra`.

I would like to see a flag which lists all the services user wanna generate. This flag will be under `generate` and not global.

so something like this


`kedge -f wordpress.yml generate -s web -s db`

**Comment by kadel** [comment]
I don't think we should do that. 
The case should be that if you don't want to everything at once put it in the separate files.
Using file per service is the best practice and we should push to that direction.

**Comment by pradeepto** [comment]
> The case should be that if you don't want to everything at once put it in separate files.
> Using file per service is the best practice and we should push to that direction.

@kadel I agree with you about the best practice part and that is what we should tell people - use single file per service. But the use case still is important to give "single file" users a better experience.

**Comment by kadel** [comment]
If a developer reached the point where he/she needs to deploy only one service from a file where multiple apps are defined then it's the point where this file should be splited into multiple once.

**Comment by surajssd** [comment]
@kadel I agree on that part this could be a good practice to enforce

**Comment by kadel** [comment]
What if we put this on the lowest priority. 
We can implement this only once it's clear this is really needed by some real use case.
